designs/0035-cmdstanrs.md

```diff
@@ -0,0 +1,348 @@
+- Feature Name: cmdstanrs
+- Start Date: 2023-11-22
+- RFC PR:
+- Stan Issue:
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a Rust interface for CmdStan through compiled
+executables, that is, no direct interface with C++.
+
+The goal is to provide an interface which enables users to:
+- compile Stan programs (with arbitrary options)
+- build and compose arguments/options (to be passed to C++
+  executables) in idiomatic Rust
+- call C++ executables, then memoize input and collect output (thereby
+  making these available for programmatic use)
+- call `diagnose` and `stansummary` tools and collect output
+
+The objective is to keep the interface as simple as possible.
+
+# Motivation
+[motivation]: #motivation
+
+Suppose that you write Rust code. Suppose that you use `Stan` for
+probabilistic programming. You have three choices for creating an
+application which utilizes both: shell scripts, introduce a scripting
+language (e.g. Python) dependency, or a Rust interface for `CmdStan`.
```

Comment:
There are a few more exotic options as well:

- A (hypothetical) Rust interface to the `stan/services` APIs
- Using Rust implementations of algorithms like NUTS and `bridgestan`, as is done in e.g. [`nutpie`](https://github.com/pymc-devs/nutpie) 

These may be worth mentioning in the Rationale and Alternatives, as both have their own downsides as well.